engine/src/main/java/org/terasology/rendering/particles/ParticleCommands.java

```diff
@@ -0,0 +1,538 @@
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector4f;
+import org.terasology.physics.HitResult;
+import org.terasology.physics.Physics;
+import org.terasology.physics.StandardCollisionGroup;
+import org.terasology.registry.In;
+import org.terasology.rendering.cameras.Camera;
+import org.terasology.rendering.particles.components.OverheadParticleSystem;
+import org.terasology.rendering.particles.components.ParticleEmitterComponent;
+import org.terasology.rendering.particles.components.ParticleSystemComponent;
+import org.terasology.rendering.particles.components.affectors.DampingAffectorComponent;
+import org.terasology.rendering.particles.components.affectors.EnergyColorAffectorComponent;
+import org.terasology.rendering.particles.components.affectors.EnergyScaleAffectorComponent;
+import org.terasology.rendering.particles.components.affectors.ForceAffectorComponent;
+import org.terasology.rendering.particles.components.affectors.ForwardEulerAffectorComponent;
+import org.terasology.rendering.particles.components.affectors.PointForceAffectorComponent;
+import org.terasology.rendering.particles.components.affectors.TurbulenceAffectorComponent;
+import org.terasology.rendering.particles.components.generators.ColorRangeGeneratorComponent;
+import org.terasology.rendering.particles.components.generators.EnergyRangeGeneratorComponent;
+import org.terasology.rendering.particles.components.generators.PositionRangeGeneratorComponent;
+import org.terasology.rendering.particles.components.generators.ScaleRangeGeneratorComponent;
+import org.terasology.rendering.particles.components.generators.TextureOffsetGeneratorComponent;
+import org.terasology.rendering.particles.components.generators.VelocityRangeGeneratorComponent;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.utilities.Assets;
+
+/**
+ * Particle related commands, mostly for testing.
+ *
+ * Created by Max Borsch on 11/28/2016.
+ */
+
+@RegisterSystem
+public class ParticleCommands extends BaseComponentSystem {
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Physics physics;
+
+    public enum FireStyle {
+        SQUARE,
+        DEFAULT,
+        SMOKEY,
+        TOON1,
+        TOON2
+    }
+
+    public EntityRef createParticleSystem() {
+        EntityBuilder psEntityBuilder = entityManager.newBuilder();
+        psEntityBuilder.setPersistent(false);
+        ParticleSystemComponent particleSystemComponent = new ParticleSystemComponent();
+        psEntityBuilder.addComponent(particleSystemComponent);
+
+        EntityRef particleSystem = psEntityBuilder.build();
+        particleSystemComponent.addEmitter(createEmitter(particleSystem));
+
+        return particleSystem;
+    }
+
+    private EntityRef createEmitter(EntityRef particleSystem) {
+        EntityBuilder emmiterEntityBuilder = entityManager.newBuilder();
```

Comment:
Nope that's the system to attach emitter to, these functions are not meant to be used or even looked at, I'm removing / replacing them later, these are hacks for debugging.